### PR TITLE
fix(codegen): ensure unique c++ variable names to avoid collision for properties with the same name

### LINKED ISSUE #53839: Codegen variable naming can result in incorrect code
### Description

```tsx
type MyEventType = {
  type: string;
  payload: {
    message: string;
  }
};

export interface NativeProps extends ViewProps {
  onPressed?: CodegenTypes.BubblingEventHandler<MyEventType> | null;
}
```

codegen generates such a code not the `payload` variable gets shadowed. And the bug is in the line: `  payload.setProperty(runtime, "payload", payload);
` 

```tsx
void SimpleViewEventEmitter::onPressed(OnPressed event) const {
  dispatchEvent("pressed", [event=std::move(event)](jsi::Runtime &runtime) {
    auto payload = jsi::Object(runtime);
    payload.setProperty(runtime, "type", event.type);
{
  auto payload = jsi::Object(runtime);
  payload.setProperty(runtime, "message", event.payload.message);
  payload.setProperty(runtime, "payload", payload);
}
    return payload;
  });
}
```

See 
https://github.com/mfazekas/codegen-variable-naming-bug

### Steps to reproduce

```
check out https://github.com/mfazekas/codegen-variable-naming-bug
cd ReproducerApp/ios
pod install
cat build/generated/ios/react/renderer/components/AppSpec/EventEmitters.cpp
```

### React Native Version

0.81.4

### Affected Platforms

Other (please specify)

### Areas

Codegen

### Output of `npx @react-native-community/cli info`

```text
info Fetching system and libraries information...
System:
  OS: macOS 15.6
  CPU: (12) arm64 Apple M2 Max
  Memory: 125.66 MB / 32.00 GB
  Shell:
    version: "5.9"
    path: /bin/zsh
Binaries:
  Node:
    version: 22.16.0
    path: ~/.nvm/versions/node/v22.16.0/bin/node
  Yarn:
    version: 1.22.21
    path: ~/.local/bin/yarn
  npm:
    version: 11.5.2
    path: ~/.nvm/versions/node/v22.16.0/bin/npm
  Watchman:
    version: 2025.06.30.00
    path: /opt/homebrew/bin/watchman
Managers:
  CocoaPods:
    version: 1.16.2
    path: /Users/boga/.rbenv/shims/pod
SDKs:
  iOS SDK:
    Platforms:
      - DriverKit 24.5
      - iOS 18.5
      - macOS 15.5
      - tvOS 18.5
      - visionOS 2.5
      - watchOS 11.5
  Android SDK: Not Found
IDEs:
  Android Studio: 2025.1 AI-251.26094.121.2513.14007798
  Xcode:
    version: 16.4/16F6
    path: /usr/bin/xcodebuild
Languages:
  Java:
    version: 20.0.1
    path: /usr/bin/javac
  Ruby:
    version: 2.7.8
    path: /Users/boga/.rbenv/shims/ruby
npmPackages:
  "@react-native-community/cli": Not Found
  react: Not Found
  react-native: Not Found
  react-native-macos: Not Found
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: true
  newArchEnabled: true
iOS:
  hermesEnabled: true
  newArchEnabled: true
```

### Stacktrace or Logs

```text
n/a
```

### MANDATORY Reproducer

 https://github.com/mfazekas/codegen-variable-naming-bug

### Screenshots and Videos

_No response_

### PR DESCRIPTION
**Pull request by mfazekas**
## Summary:

Fixes: #53839

Codegen uses object propertyNames as variable names. This breaks if the same property names appears at multiple levels. Also breaks if a property is called `payload` as payload is a variable name already used.

Before the fix codegen could generate code like this:
```tsx
payload.setProperty(runtime, "payload", payload);
```

## Changelog:

[GENERAL] [FIXED] - Fix Codegen to use unique variable names to avoid name collisions in corner cases

## Test Plan:

- jest snapshot tests

### CODE REVIEW

File: packages/react-native-codegen/e2e/deep_imports/__tests__/components/__snapshots__/GenerateEventEmitterCpp-test.js.snap [hunk 1]
```diff
@@ -153,29 +153,29 @@ void EventNestedObjectPropsNativeComponentViewEventEmitter::onChange(OnChange ev
   dispatchEvent(\\"change\\", [event=std::move(event)](jsi::Runtime &runtime) {
     auto payload = jsi::Object(runtime);
     {
-  auto location = jsi::Object(runtime);
+  auto location_1 = jsi::Object(runtime);
```
Comment (cortinico) [inline packages/react-native-codegen/e2e/deep_imports/__tests__/components/__snapshots__/GenerateEventEmitterCpp-test.js.snap]: maybe let's suffix them like `location_safe` or `location_codegen`?

I think that having numeric suffixes becomes really impractical really quickly.

Comment (mfazekas) [reply packages/react-native-codegen/e2e/deep_imports/__tests__/components/__snapshots__/GenerateEventEmitterCpp-test.js.snap]: I've changed it to `location_codegen1`. The numbers are needed so 

```
type MyEventType = {
  type: string;
  payload: {
    message: string;
    body: {
      subject: string;
      body: {
        title: string;
        description: string;
      };
    }
    foo: {
      foo: string[];
    }
    bar: {
      bar: {
        a: string;
      }
    }
  }
};

export interface NativeProps extends ViewProps {
  onPressed?: CodegenTypes.BubblingEventHandler<MyEventType> | null;
}
```

because right now the above structure generates
```c++
      body.setProperty(runtime, "body", body);
      ...
        foo.setProperty(runtime, "foo", foo);
      ...
            bar.setProperty(runtime, "bar", bar);

     payload.setProperty(runtime, "payload", payload);
``` 

with suffixes we generate:
```
      body_codegen2.setProperty(runtime, "body", body_codegen3);
        ...
        foo_codegen4.setProperty(runtime, "foo", foo_codegen5);
        ...
              bar_codegen8.setProperty(runtime, "bar", bar_codegen9);
        ...      
          payload.setProperty(runtime, "payload", payload_codegen1);
```

**Review by note89304214615-hue — COMMENTED** [review]
Qu

**Comment by javache** [comment]
Thanks for the fix! Could we make this even more explicit by using variable names such as `__codegen__location`

**Comment by cortinico** [comment]
> Thanks for the fix! Could we make this even more explicit by using variable names such as `__codegen__location`

cc @mfazekas

**Comment by mfazekas** [comment]
> > Thanks for the fix! Could we make this even more explicit by using variable names such as `__codegen__location`
> 
> cc @mfazekas

Just to clarify if name is `location` right now should it be
`__codegen__2__location`
